core/src/main/java/org/pentaho/di/compatibility/ValueDate.java

```diff
@@ -0,0 +1,189 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2017 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.compatibility;
+
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.pentaho.di.core.Const;
+
+/**
+ * This class contains a Value of type Date.
+ *
+ * @author Matt
+ * @since 15-10-2004
+ *
+ */
+public class ValueDate implements ValueInterface, Cloneable {
+  public static final String DATE_FORMAT = "yyyy/MM/dd HH:mm:ss.SSS";
+  private Date date;
+  public int precision;
+
+  public ValueDate() {
+    this.date = null;
+    this.precision = -1;
+  }
+
+  public ValueDate( Date date ) {
+    this.date = date;
+    this.precision = -1;
+  }
+
+  @Override
+  public int getType() {
+    return Value.VALUE_TYPE_DATE;
+  }
+
+  @Override
+  public String getTypeDesc() {
+    return "Date";
+  }
+
+  @Override
+  public String getString() {
+    if ( date == null ) {
+      return null;
+    }
+    SimpleDateFormat df = new SimpleDateFormat( DATE_FORMAT );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.